File: Modelo/ItemCarrito.cs
using System;

namespace punto_de_venta_C_.Modelo
{
    public class ItemCarrito
    {
        private Producto producto;
        private double cantidad;

        public ItemCarrito(Producto producto, double cantidad)
        {
            this.producto = producto;
            this.cantidad = cantidad;
        }

        public Producto getProducto() { return producto; }
        public void setProducto(Producto producto) { this.producto = producto; }
        public double getCantidad() { return cantidad; }
        public void setCantidad(double cantidad)
        {
            this.cantidad = Math.Round(cantidad * 1000.0) / 1000.0;
        }

        public double getSubtotal()
        {
            return Math.Round(producto.getPrecio() * cantidad * 100.0) / 100.0;
        }

        public void incrementar(double delta)
        {
            this.cantidad = Math.Round((this.cantidad + delta) * 1000.0) / 1000.0;
        }

        public void decrementar(double delta)
        {
            double nueva = this.cantidad - delta;
            this.cantidad = Math.Max(0, Math.Round(nueva * 1000.0) / 1000.0);
        }
    }
}

File: Modelo/ItemCarrito.py
class ItemCarrito:
    def __init__(self, producto=None, cantidad=0.0):
        self._producto = producto
        self._cantidad = cantidad

    def getProducto(self):
        return self._producto

    def setProducto(self, producto):
        self._producto = producto

    def getCantidad(self):
        return self._cantidad

    def setCantidad(self, cantidad):
        self._cantidad = round(cantidad, 3)

    def getSubtotal(self):
        if self._producto is None:
            return 0.0
        return round(self._producto.getPrecio() * self._cantidad, 2)

    def incrementar(self, delta):
        self._cantidad = round(self._cantidad + delta, 3)

    def decrementar(self, delta):
        nueva = self._cantidad - delta
        self._cantidad = max(0.0, round(nueva, 3))